### ISSUE TITLE
ERFNet-CULane-PyTorch training issue

### ISSUE THREAD

**Issue by yoga-0125** [cardwing/Codes-for-Lane-Detection#268]
With default hyper-parameters and TuSimple dataset:

1. test (validation) result only shows iterations according to [validation data size 358 / batch size 8 = 45] whatever the epoch value is set (attached). Why epoch number has no any influence in training?

2. After training, no any checkpoint files are saved into the "trained" folder, I guess the error comes from the epoch problem. Any advice? 

=> loading checkpoint './pretrained/ERFNet_pretrained.tar'
#reused param: 302
=> loaded checkpoint 'False' (epoch 16)
/usr/local/lib/python3.6/dist-packages/torch/nn/functional.py:1569: UserWarning: nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.
  warnings.warn("nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.")
Test: [0/45]	Time 5.703 (5.703)	Loss 1.4794 (1.4794)	Pixels Acc 0.598	mIoU 0.108
Test: [1/45]	Time 0.617 (3.160)	Loss 1.3363 (1.4078)	Pixels Acc 0.626	mIoU 0.112
Test: [2/45]	Time 1.217 (2.512)	Loss 1.3125 (1.3761)	Pixels Acc 0.640	mIoU 0.114
Test: [3/45]	Time 0.704 (2.060)	Loss 1.3707 (1.3747)	Pixels Acc 0.639	mIoU 0.114
Test: [4/45]	Time 0.968 (1.842)	Loss 1.3242 (1.3646)	Pixels Acc 0.641	mIoU 0.115
Test: [5/45]	Time 0.716 (1.654)	Loss 1.4774 (1.3834)	Pixels Acc 0.635	mIoU 0.114
Test: [6/45]	Time 0.762 (1.527)	Loss 1.4885 (1.3984)	Pixels Acc 0.633	mIoU 0.114
Test: [7/45]	Time 0.369 (1.382)	Loss 1.4809 (1.4087)	Pixels Acc 0.626	mIoU 0.112
Test: [8/45]	Time 0.317 (1.264)	Loss 1.4973 (1.4186)	Pixels Acc 0.620	mIoU 0.111
Test: [9/45]	Time 0.302 (1.168)	Loss 1.4334 (1.4200)	Pixels Acc 0.618	mIoU 0.111
Test: [10/45]	Time 0.293 (1.088)	Loss 1.4384 (1.4217)	Pixels Acc 0.617	mIoU 0.111
Test: [11/45]	Time 0.298 (1.022)	Loss 1.4108 (1.4208)	Pixels Acc 0.619	mIoU 0.111
Test: [12/45]	Time 0.331 (0.969)	Loss 1.4783 (1.4252)	Pixels Acc 0.619	mIoU 0.111
Test: [13/45]	Time 0.292 (0.921)	Loss 1.4863 (1.4296)	Pixels Acc 0.619	mIoU 0.112
Test: [14/45]	Time 0.152 (0.869)	Loss 1.4116 (1.4284)	Pixels Acc 0.618	mIoU 0.111
Test: [15/45]	Time 0.148 (0.824)	Loss 1.3707 (1.4248)	Pixels Acc 0.619	mIoU 0.111
Test: [16/45]	Time 0.147 (0.784)	Loss 1.4696 (1.4274)	Pixels Acc 0.618	mIoU 0.111
Test: [17/45]	Time 0.156 (0.750)	Loss 1.4729 (1.4299)	Pixels Acc 0.617	mIoU 0.111
Test: [18/45]	Time 0.149 (0.718)	Loss 1.4157 (1.4292)	Pixels Acc 0.618	mIoU 0.111
Test: [19/45]	Time 0.146 (0.689)	Loss 1.3357 (1.4245)	Pixels Acc 0.619	mIoU 0.111
Test: [20/45]	Time 0.148 (0.664)	Loss 1.3366 (1.4203)	Pixels Acc 0.621	mIoU 0.112
Test: [21/45]	Time 0.146 (0.640)	Loss 1.3156 (1.4156)	Pixels Acc 0.623	mIoU 0.112
Test: [22/45]	Time 0.148 (0.619)	Loss 1.3511 (1.4128)	Pixels Acc 0.623	mIoU 0.112
Test: [23/45]	Time 0.144 (0.599)	Loss 1.3298 (1.4093)	Pixels Acc 0.624	mIoU 0.112
Test: [24/45]	Time 0.143 (0.581)	Loss 1.3089 (1.4053)	Pixels Acc 0.626	mIoU 0.112
Test: [25/45]	Time 0.149 (0.564)	Loss 1.2821 (1.4006)	Pixels Acc 0.627	mIoU 0.113
Test: [26/45]	Time 0.147 (0.549)	Loss 1.4162 (1.4011)	Pixels Acc 0.627	mIoU 0.113
Test: [27/45]	Time 0.147 (0.534)	Loss 1.4246 (1.4020)	Pixels Acc 0.625	mIoU 0.112
Test: [28/45]	Time 0.147 (0.521)	Loss 1.4109 (1.4023)	Pixels Acc 0.624	mIoU 0.112
Test: [29/45]	Time 0.143 (0.508)	Loss 1.2998 (1.3989)	Pixels Acc 0.626	mIoU 0.112
Test: [30/45]	Time 0.144 (0.497)	Loss 1.3755 (1.3981)	Pixels Acc 0.625	mIoU 0.112
Test: [31/45]	Time 0.144 (0.486)	Loss 1.3729 (1.3973)	Pixels Acc 0.625	mIoU 0.112
Test: [32/45]	Time 0.147 (0.475)	Loss 1.3714 (1.3965)	Pixels Acc 0.626	mIoU 0.112
Test: [33/45]	Time 0.151 (0.466)	Loss 1.4971 (1.3995)	Pixels Acc 0.625	mIoU 0.112
Test: [34/45]	Time 0.151 (0.457)	Loss 1.3613 (1.3984)	Pixels Acc 0.625	mIoU 0.112
Test: [35/45]	Time 0.144 (0.448)	Loss 1.3291 (1.3965)	Pixels Acc 0.627	mIoU 0.112
Test: [36/45]	Time 0.143 (0.440)	Loss 1.3475 (1.3952)	Pixels Acc 0.627	mIoU 0.112
Test: [37/45]	Time 0.146 (0.432)	Loss 1.4607 (1.3969)	Pixels Acc 0.627	mIoU 0.112
Test: [38/45]	Time 0.144 (0.425)	Loss 1.3989 (1.3969)	Pixels Acc 0.627	mIoU 0.112
Test: [39/45]	Time 0.146 (0.418)	Loss 1.3811 (1.3965)	Pixels Acc 0.626	mIoU 0.112
Test: [40/45]	Time 0.152 (0.411)	Loss 1.3961 (1.3965)	Pixels Acc 0.626	mIoU 0.112
Test: [41/45]	Time 0.154 (0.405)	Loss 1.4259 (1.3972)	Pixels Acc 0.625	mIoU 0.112
Test: [42/45]	Time 0.147 (0.399)	Loss 1.2867 (1.3947)	Pixels Acc 0.626	mIoU 0.112
Test: [43/45]	Time 0.148 (0.393)	Loss 1.4124 (1.3951)	Pixels Acc 0.626	mIoU 0.112
Test: [44/45]	Time 0.112 (0.387)	Loss 1.2874 (1.3933)	Pixels Acc 0.627	mIoU 0.112
Testing Results: Pixels Acc 0.627	mIoU 0.112 (0.1123)	Loss 1.39326

**Comment by cardwing** [comment]
The checkpoint is saved in each epoch. Maybe the error is caused by the training process. Please debug by yourself.

**Comment by foxkw** [comment]
> With default hyper-parameters and TuSimple dataset:
> 
>     1. test (validation) result only shows iterations according to [validation data size 358 / batch size 8 = 45] whatever the epoch value is set (attached). Why epoch number has no any influence in training?
> 
>     2. After training, no any checkpoint files are saved into the "trained" folder, I guess the error comes from the epoch problem. Any advice?
> 
> 
> => loading checkpoint './pretrained/ERFNet_pretrained.tar'
> #reused param: 302
> => loaded checkpoint 'False' (epoch 16)
> /usr/local/lib/python3.6/dist-packages/torch/nn/functional.py:1569: UserWarning: nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.
> warnings.warn("nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.")
> Test: [0/45] Time 5.703 (5.703) Loss 1.4794 (1.4794) Pixels Acc 0.598 mIoU 0.108
> Test: [1/45] Time 0.617 (3.160) Loss 1.3363 (1.4078) Pixels Acc 0.626 mIoU 0.112
> Test: [2/45] Time 1.217 (2.512) Loss 1.3125 (1.3761) Pixels Acc 0.640 mIoU 0.114
> Test: [3/45] Time 0.704 (2.060) Loss 1.3707 (1.3747) Pixels Acc 0.639 mIoU 0.114
> Test: [4/45] Time 0.968 (1.842) Loss 1.3242 (1.3646) Pixels Acc 0.641 mIoU 0.115
> Test: [5/45] Time 0.716 (1.654) Loss 1.4774 (1.3834) Pixels Acc 0.635 mIoU 0.114
> Test: [6/45] Time 0.762 (1.527) Loss 1.4885 (1.3984) Pixels Acc 0.633 mIoU 0.114
> Test: [7/45] Time 0.369 (1.382) Loss 1.4809 (1.4087) Pixels Acc 0.626 mIoU 0.112
> Test: [8/45] Time 0.317 (1.264) Loss 1.4973 (1.4186) Pixels Acc 0.620 mIoU 0.111
> Test: [9/45] Time 0.302 (1.168) Loss 1.4334 (1.4200) Pixels Acc 0.618 mIoU 0.111
> Test: [10/45] Time 0.293 (1.088) Loss 1.4384 (1.4217) Pixels Acc 0.617 mIoU 0.111
> Test: [11/45] Time 0.298 (1.022) Loss 1.4108 (1.4208) Pixels Acc 0.619 mIoU 0.111
> Test: [12/45] Time 0.331 (0.969) Loss 1.4783 (1.4252) Pixels Acc 0.619 mIoU 0.111
> Test: [13/45] Time 0.292 (0.921) Loss 1.4863 (1.4296) Pixels Acc 0.619 mIoU 0.112
> Test: [14/45] Time 0.152 (0.869) Loss 1.4116 (1.4284) Pixels Acc 0.618 mIoU 0.111
> Test: [15/45] Time 0.148 (0.824) Loss 1.3707 (1.4248) Pixels Acc 0.619 mIoU 0.111
> Test: [16/45] Time 0.147 (0.784) Loss 1.4696 (1.4274) Pixels Acc 0.618 mIoU 0.111
> Test: [17/45] Time 0.156 (0.750) Loss 1.4729 (1.4299) Pixels Acc 0.617 mIoU 0.111
> Test: [18/45] Time 0.149 (0.718) Loss 1.4157 (1.4292) Pixels Acc 0.618 mIoU 0.111
> Test: [19/45] Time 0.146 (0.689) Loss 1.3357 (1.4245) Pixels Acc 0.619 mIoU 0.111
> Test: [20/45] Time 0.148 (0.664) Loss 1.3366 (1.4203) Pixels Acc 0.621 mIoU 0.112
> Test: [21/45] Time 0.146 (0.640) Loss 1.3156 (1.4156) Pixels Acc 0.623 mIoU 0.112
> Test: [22/45] Time 0.148 (0.619) Loss 1.3511 (1.4128) Pixels Acc 0.623 mIoU 0.112
> Test: [23/45] Time 0.144 (0.599) Loss 1.3298 (1.4093) Pixels Acc 0.624 mIoU 0.112
> Test: [24/45] Time 0.143 (0.581) Loss 1.3089 (1.4053) Pixels Acc 0.626 mIoU 0.112
> Test: [25/45] Time 0.149 (0.564) Loss 1.2821 (1.4006) Pixels Acc 0.627 mIoU 0.113
> Test: [26/45] Time 0.147 (0.549) Loss 1.4162 (1.4011) Pixels Acc 0.627 mIoU 0.113
> Test: [27/45] Time 0.147 (0.534) Loss 1.4246 (1.4020) Pixels Acc 0.625 mIoU 0.112
> Test: [28/45] Time 0.147 (0.521) Loss 1.4109 (1.4023) Pixels Acc 0.624 mIoU 0.112
> Test: [29/45] Time 0.143 (0.508) Loss 1.2998 (1.3989) Pixels Acc 0.626 mIoU 0.112
> Test: [30/45] Time 0.144 (0.497) Loss 1.3755 (1.3981) Pixels Acc 0.625 mIoU 0.112
> Test: [31/45] Time 0.144 (0.486) Loss 1.3729 (1.3973) Pixels Acc 0.625 mIoU 0.112
> Test: [32/45] Time 0.147 (0.475) Loss 1.3714 (1.3965) Pixels Acc 0.626 mIoU 0.112
> Test: [33/45] Time 0.151 (0.466) Loss 1.4971 (1.3995) Pixels Acc 0.625 mIoU 0.112
> Test: [34/45] Time 0.151 (0.457) Loss 1.3613 (1.3984) Pixels Acc 0.625 mIoU 0.112
> Test: [35/45] Time 0.144 (0.448) Loss 1.3291 (1.3965) Pixels Acc 0.627 mIoU 0.112
> Test: [36/45] Time 0.143 (0.440) Loss 1.3475 (1.3952) Pixels Acc 0.627 mIoU 0.112
> Test: [37/45] Time 0.146 (0.432) Loss 1.4607 (1.3969) Pixels Acc 0.627 mIoU 0.112
> Test: [38/45] Time 0.144 (0.425) Loss 1.3989 (1.3969) Pixels Acc 0.627 mIoU 0.112
> Test: [39/45] Time 0.146 (0.418) Loss 1.3811 (1.3965) Pixels Acc 0.626 mIoU 0.112
> Test: [40/45] Time 0.152 (0.411) Loss 1.3961 (1.3965) Pixels Acc 0.626 mIoU 0.112
> Test: [41/45] Time 0.154 (0.405) Loss 1.4259 (1.3972) Pixels Acc 0.625 mIoU 0.112
> Test: [42/45] Time 0.147 (0.399) Loss 1.2867 (1.3947) Pixels Acc 0.626 mIoU 0.112
> Test: [43/45] Time 0.148 (0.393) Loss 1.4124 (1.3951) Pixels Acc 0.626 mIoU 0.112
> Test: [44/45] Time 0.112 (0.387) Loss 1.2874 (1.3933) Pixels Acc 0.627 mIoU 0.112
> Testing Results: Pixels Acc 0.627 mIoU 0.112 (0.1123) Loss 1.39326

Hi , Have you solved this problem . I met the same problem, mIOU remained at 0.044 , epoch doesn't work
